### PR TITLE
Make `flags` an alpha-sorted string

### LINKED ISSUE #11: Make flags an alpha-sorted string
Currently flags is an array, we should make it as an alphabetically sorted string, as in [RegExp.prototype.flags](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/flags).

```js
regexpTree.parse('/a/ig');
```

Should be:

```js
{
  "type": "RegExp",
  "body": {
    "type": "Char",
    "value": "a",
    "kind": "simple"
  },
  "flags": "gi"
}
```

### PR DESCRIPTION
**Pull request by shvaikalesh**
Closes #11.

### CODE REVIEW

File: src/parser/__tests__/parser-basic-test.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ describe('basic', () => {
         value: 'a',
         kind: 'simple'
       },
-      flags: []
+      flags: '',
```
Comment (DmitrySoshnikov) [inline src/parser/__tests__/parser-basic-test.js]: I'm wondering whether `null` would be better in case of no flags?

Comment (shvaikalesh) [inline src/parser/__tests__/parser-basic-test.js]: I think empty string is better.

Most common usages: check whether flag is enabled, remove flag, add flag. All three would require null checks (meh) and docs pointing that out. Furthermore, we are mimicking `RegExp#flags`, it returns an empty string.

**Comment by shvaikalesh** [comment]
README is what I always forget, done.

File: src/parser/__tests__/parser-basic-test.js (hunk 1 above)
Comment (DmitrySoshnikov) [inline src/parser/__tests__/parser-basic-test.js]: Yeah, good point, OK.

**Comment by DmitrySoshnikov** [comment]
Cool, there is [one more](https://github.com/DmitrySoshnikov/regexp-tree/blob/master/src/parser/README.md) README in the parser directory.

**Comment by shvaikalesh** [comment]
My bad, fixed that.

**Comment by DmitrySoshnikov** [comment]
Nice! 👍

**Pull request merged**